File: Introduction to DSA/convertDecimalToBinary.cpp
#include <iostream>
using namespace std;

string convertDecimalToBinary(int decimalNumber){
    string binaryNumber = "";
    while(decimalNumber!=0){
        int remainder = decimalNumber%2;
        binaryNumber = to_string(remainder) + binaryNumber;
        decimalNumber = decimalNumber/2;
    }
    return binaryNumber;
}

int main(){
    int decimalNumber;
    cout<<"Enter the decimal number: ";
    cin>>decimalNumber;

    if(decimalNumber!=0 && decimalNumber>0){
        string binaryNumber = convertDecimalToBinary(decimalNumber);
        cout<<"The Binary Number for "<<decimalNumber<<" is "<<binaryNumber<<endl;
    }
    else if(decimalNumber==0){
        cout<<"The Binary Number for "<<decimalNumber<<" is 0"<<endl;
    }
    else{
        cout<<"Enter a valid input"<<endl;
    }

    return 0;
}

File: Introduction to DSA/convert_decimal_to_binary.py
decimal = int(input("Enter a decimanl number to convert it to binary: "))
dec_copy = decimal
remainders = []
while decimal!=0:
  remainders.append(decimal%2)
  decimal = decimal // 2

print(f"Decimanl ({dec_copy}): Binary({int("".join(map(str, remainders[::-1])))})")